addons/l10n_sa_edi/tests/common.py

```diff
@@ -53,9 +53,20 @@ def _setup_company(cls):
             'l10n_sa_edi_plot_identification': '1234',
             'l10n_sa_additional_identification_number': '2525252525252',
             'l10n_sa_additional_identification_scheme': 'CRN',  # Commercial Registration Number
-        })
+            **(defaults or {})
+        }
+
+    @classmethod
+    def _setup_company(cls):
+        """Configure the test company with Saudi Arabia specific settings."""
+        cls.company.write(cls._get_company_vals())
         cls.company.l10n_sa_private_key = cls.env['res.company']._l10n_sa_generate_private_key()
 
+    @classmethod
+    def _setup_branches(cls):
+        vals = cls._get_company_vals({"name": "SA Branch", "parent_id": cls.company.id})
+        cls.sa_branch = cls.env["res.company"].create(vals)
```

Comment:
Why not just keep the same thing as before ?
```suggestion
        cls.sa_branch = cls.env["res.company"].create({
            'name': 'SA Branch',
            'parent_id': cls.company.id,
            'country_id': cls.company.country_id.id,
        })
```

Reply:
we need the tax and address information from _get_company_vals to be able to post the invoices without issues, so we created this helper